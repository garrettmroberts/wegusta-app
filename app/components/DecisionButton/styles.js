import { StyleSheet } from 'react-native';
import { colors } from '../../config';

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.white,
    height: 80,
    width: 80,
    borderRadius: 90,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: colors.primary,
    shadowOffset: {
      width: 0,
      height: 3
    },
    shadowOpacity: 0.29,
    shadowRadius: 4.65,
    elevation: 7
  },
  icon: {
    height: 32,
    width: 32,
    resizeMode: 'contain'
  }
});

export default styles;
