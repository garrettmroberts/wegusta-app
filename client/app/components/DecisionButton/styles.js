import { StyleSheet } from 'react-native'
import { colors } from '../../config'

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.white,
    height: 80,
    width: 80,
    borderRadius: 90,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: colors.primary,
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.29,
    shadowRadius: 4.65,
    elevation: 7,
  },
  icon: {
    position: 'relative',
    top: 2,
  },
})

export default styles
