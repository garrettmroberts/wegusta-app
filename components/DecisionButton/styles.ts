import { StyleSheet } from 'react-native'
import Colors from '../../constants/Colors'

const styles = StyleSheet.create({
  button: {
    height: 80,
    width: 80,
    borderRadius: 90,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  like: {
    backgroundColor: `${Colors.error}20`
  },
  dislike: {
    backgroundColor: Colors.greyLight
  },
  icon: {
    position: 'relative',
    top: 2
  }
})

export default styles
