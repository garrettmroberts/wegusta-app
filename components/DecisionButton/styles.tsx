import { StyleSheet } from 'react-native';
import Colors from '../../constants/Colors';

const styles = StyleSheet.create({
  button: {
    backgroundColor: Colors.white,
    height: 80,
    width: 80,
    borderRadius: 90,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: Colors.primary,
    shadowOffset: {
      width: 0,
      height: 3
    },
    shadowOpacity: 0.29,
    shadowRadius: 4.65,
    elevation: 7
  },
  icon: {
    position: 'relative',
    top: 2
  }
});

export default styles;
