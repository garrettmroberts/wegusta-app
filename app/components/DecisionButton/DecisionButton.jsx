import React from 'react';
import { Pressable, Image } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

const icon = decision => {
  if (decision === 'like') {
    return(<Image source={require('../../assets/heart-icon.png')} style={styles.icon} />);
  } else {
    return(<Image source={require('../../assets/x-icon.png')} style={styles.icon} />);
  }
};

const DecisionButton = ({ decision, handlePress }) => {
  return(
    <Pressable style={ styles.button } onPress={ handlePress } >
      { icon(decision) }
    </Pressable>
  );
};

DecisionButton.propTypes = {
  decision: PropTypes.string.isRequired,
  handlePress: PropTypes.func
};

export { DecisionButton };
