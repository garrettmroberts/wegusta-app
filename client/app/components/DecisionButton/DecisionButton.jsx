import React from 'react'
import { Pressable, View } from 'react-native'
import PropTypes from 'prop-types'

import styles from './styles'
import { colors } from '../../config'
import buildIcon from '../../utils/buildIcon'

const DecisionButton = ({ decision, handlePress }) => {
    const icon = decision => {
        if (decision === 'like') {
            return buildIcon({ name: 'heart', size: 50, color: colors.error })
        } else {
            return buildIcon({ name: 'close', size: 60, color: colors.primary })
        }
    }

    return (
        <Pressable
            style={({ pressed }) => [
                styles.button,
                { opacity: pressed ? 0.8 : 1 },
            ]}
            onPress={handlePress}
        >
            <View style={styles.icon}>{icon(decision)}</View>
        </Pressable>
    )
}

DecisionButton.propTypes = {
    decision: PropTypes.string.isRequired, // like, dislike
    handlePress: PropTypes.func,
}

export { DecisionButton }
