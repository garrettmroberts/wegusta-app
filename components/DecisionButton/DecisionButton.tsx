import React from 'react'
import { Pressable, View } from 'react-native'

import styles from './styles'
import { Ionicons } from '@expo/vector-icons'
import Colors from '../../constants/Colors'

type Props = {
  decision: 'like' | 'dislike',
  handlePress?: () => {}
}

const DecisionButton = ({ decision, handlePress}: Props) => {
  const icon = (decision: 'like' | 'dislike') => {
    if (decision === 'like') {
      return <Ionicons name='heart' size={50} color={Colors.error} />
    } else {
      return <Ionicons name='close' size={60} color={Colors.primary} />
    }
  }

  return (
    <Pressable
      style={({ pressed }) => [styles.button, { opacity: pressed ? 0.8 : 1 }]}
      onPress={handlePress}
    >
      <View style={styles.icon}>{icon(decision)}</View>
    </Pressable>
  )
}

export default DecisionButton