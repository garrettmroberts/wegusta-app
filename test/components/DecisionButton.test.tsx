import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';

import DecisionButton from '../../components/DecisionButton/DecisionButton';

describe('<DecisionButton />', () => {
  it('runs handlePress function on press', () => {
    const mockFn = jest.fn();
    render(<DecisionButton decision="like" onPress={mockFn} />);
    fireEvent.press(screen.getByTestId('decision-button'));
    expect(mockFn).toBeCalled();
  });

  it('renders a like button correctly', () => {
    const { toJSON } = render(<DecisionButton decision="like" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders a dislike button correctly', () => {
    const { toJSON } = render(<DecisionButton decision="dislike" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
