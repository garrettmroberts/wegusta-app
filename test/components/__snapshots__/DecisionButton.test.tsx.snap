// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DecisionButton /> renders a dislike button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 90,
        "display": "flex",
        "elevation": 7,
        "height": 80,
        "justifyContent": "center",
        "shadowColor": "#4667E8",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.29,
        "shadowRadius": 4.65,
        "width": 80,
      },
      Object {
        "opacity": 1,
      },
    ]
  }
  testID="decision-button"
>
  <View
    style={
      Object {
        "position": "relative",
        "top": 2,
      }
    }
  >
    <View
      color="#4667E8"
      name="close"
      size={60}
    />
  </View>
</View>
`;

exports[`<DecisionButton /> renders a like button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 90,
        "display": "flex",
        "elevation": 7,
        "height": 80,
        "justifyContent": "center",
        "shadowColor": "#4667E8",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.29,
        "shadowRadius": 4.65,
        "width": 80,
      },
      Object {
        "opacity": 1,
      },
    ]
  }
  testID="decision-button"
>
  <View
    style={
      Object {
        "position": "relative",
        "top": 2,
      }
    }
  >
    <View
      color="#FF401A"
      name="heart"
      size={50}
    />
  </View>
</View>
`;
